compiler/src/iree/compiler/Codegen/Common/FlattenMemRefSubspanPass.cpp

```diff
@@ -602,14 +602,21 @@ struct FoldSubspanOffsetIntoLoadStore final : public OpRewritePattern<OpType> {
 
   LogicalResult matchAndRewrite(OpType op,
                                 PatternRewriter &rewriter) const override {
-    auto memrefType = op.getMemref().getType().template cast<MemRefType>();
+    Value memref;
+    if constexpr (std::is_same<OpType, gpu::SubgroupMmaLoadMatrixOp>::value)
```

Comment:
Nit: wrap these if-else cases with `{ ... }`. (In IREE we require that, which is kinda different than LLVM/MLIR..)